### ISSUE TITLE
Allow more algorithm to use prior knowledge

### ISSUE THREAD

**Issue by xwbxxx** [huawei-noah/trustworthyAI#79]
Just like the example of PC algorithm, I hope you can provide interfaces to allow more algorithms to adopt a priori knowledge.
![image](https://user-images.githubusercontent.com/49603007/192692097-bba01b21-904a-44d9-82e4-47e9f27025c2.png)
[PC demo.py](https://github.com/huawei-noah/trustworthyAI/blob/1868f3a3c15b9e6131263a8c9a1ccb1f59b3c045/gcastle/example/pc/pc_demo.py)

**Comment by shaido987** [comment]
Hello,

Prior knowledge is something we plan to include in all algorithms in the future (as also mentioned in #72). However, there is no timeline for this as of yet.

**Comment by AlxndrMlk** [comment]
@shaido987, is there a way to help with this task as a contributor?

**Comment by shaido987** [comment]
@AlxndrMlk Any contributions in this direction are very welcome! Prior knowledge can be a bit more difficult to integrate into some of the algorithms as compared with PC. If you want to help you can add it to one or multiple algorithms directly for now, and then we can see if the logic can be abstracted into the base class. Or if you have some suggestions in that direction we can look over them.

**Comment by AlxndrMlk** [comment]
@shaido987 

It seems that there were some attempts to introduce the possibility for adding causal knowledge to `DirectLiNGAM` (https://github.com/huawei-noah/trustworthyAI/blob/master/gcastle/castle/algorithms/lingam/direct_lingam.py), but it seems it does not work in the current version (1.0.3).

I am referring to these (lines 76-77):

```python
@check_args_value(DIRECT_LINGAM_VALID_PARAMS)
def __init__(self, prior_knowledge=None, measure='pwling', thresh=0.3):
   ...
```

I tried adding a priori knowledge to Direct LiNGAM and it did not work for me (the model was not adding the edges passed to `prior_knowlede`)

I think that actually it is possible to add the prior knowledge functionality for Direct LiNGAM (it was described in the paper if I'm not mistaken).

Perhaps that would be a good first step for enhancing the palette of models that accept prior knowledge.

What are your thoughts on this?

**Comment by shaido987** [comment]
Yes, I agree that making sure that it works for DirectLiNGAM would be a good first step forward. 

For the a prior knowledge added to the PC implementation (added in #73) we added a [`PrioriKnowledge`](https://github.com/huawei-noah/trustworthyAI/blob/master/gcastle/castle/common/priori_knowledge.py) class that can be used here. I believe it should have the same characteristics as the input matrix in DirectLiNGAM.
